csharp/src/Ice/ConnectionFactory.cs

```diff
@@ -188,17 +188,16 @@ public void SetRouterInfo(RouterInfo routerInfo)
                 // connections, so that callbacks from the router can be
                 // received over such connections.
                 //
-                DefaultsAndOverrides defaultsAndOverrides = _communicator.DefaultsAndOverrides;
                 for (int i = 0; i < endpoints.Length; i++)
                 {
                     Endpoint endpoint = endpoints[i];
 
                     //
                     // Modify endpoints with overrides.
                     //
-                    if (defaultsAndOverrides.OverrideTimeout)
+                    if (_communicator.OverrideTimeout != null)
                     {
-                        endpoint = endpoint.NewTimeout(defaultsAndOverrides.OverrideTimeoutValue);
+                        endpoint = endpoint.NewTimeout(_communicator.OverrideTimeout.Value);
                     }
```

Comment:
```
if (_communicator.OverrideTimeout is int overrideTimeout)
{
    endpoint = endpoint.NewTimeout(overrideTimeout);
}
```

Reply:
I know that's the preferred syntax in C#, but I really don't feel like it's worth leaking a temporary variable into enclosing scope, and the extra code it'll generate just to save a `.Value` access: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0ATEBqAPgWACgBiAOwFcAbSgQ2EpgAIZS6GiABAJgEYjOAzI27CeAdiIBvAJBFp3LnMlFGqtaoCWpAC4B+RgGcIAWxjaAFloDmjALyMeXANwr1a129EBOABQAiW0Cg4L8AShdCT1UPNw0AM0YfI1MLa0YNA3SdRjiICFCY9WVIqLcOHl8/AHdzGm0YADcYKD9GbBy88PcSzwBfQu7S8srg0dswiNKBzQSkkzNLUhsAQnsKagKet2LSsor/GrrG5tb25IXrADoANRpKchgu6K31fpfBqOH/MZDw6cZ/rkIHZGABWSaffZAv6EaRvXpAA
Or are there any benefits to it that I'm not seeing?

Should I still just switch to the `foo is T bar` anyways just because of it's the preferred syntax you think?